pyuvdata/uvdata/uvdata.py

```diff
@@ -5243,49 +5155,101 @@ def _apply_w_proj(self, new_w_vals, old_w_vals, select_mask=None):
+        """
+        warnings.warn(
+            "This method is deprecated in favor of `unproject_phase`, which better "
+            "describes what is being done. This method will be removed in version 2.4.",
+            DeprecationWarning,
+        )
+        self.unproject_phase(
+            phase_frame=phase_frame, use_ant_pos=use_ant_pos, use_old_proj=use_old_proj
+        )
+
+    def unproject_phase(
         self,
         phase_frame=None,
         use_ant_pos=True,
         use_old_proj=False,
+        select_mask=None,
+        cat_name="unprojected",
     ):
         """
-        Convert from a phased dataset to a drift dataset.
+        Undo phasing to get back to an `unprojected` state.
 
         See the phasing memo under docs/references for more documentation.
 
         Parameters
         ----------
         phase_frame : str
-            The astropy frame to phase from. Either 'icrs' or 'gcrs'.
-            'gcrs' accounts for precession & nutation, 'icrs' also includes abberation.
-            Defaults to using the 'phase_center_frame' attribute or 'icrs'
-            if that attribute is None.
+            Deprecated, will be removed in version 2.4. Only used if use_old_proj is
+            True. The astropy frame to phase from, either 'icrs' or 'gcrs'. 'gcrs'
+            accounts for precession & nutation, 'icrs' also includes abberation.
+            Defaults to using the "cat_frame" value in the `phase_center_catalog`.
         use_ant_pos : bool
             If True, calculate the uvws directly from the antenna positions
             rather than from the existing uvws. Default is True.
         use_old_proj : bool
-            If True, uses the 'old' way of calculating baseline projections.
-            Default is False.
+            Deprecated, will be removed in version 2.4. If True, uses the 'old' way of
+            calculating baseline projections. If a dataset is known to be phased using
+            the old way, use the `fix_phase` method to fix the phasing before calling
+            this method.
+        select_mask : ndarray of bool
+            Optional mask for selecting which data to operate on along the blt-axis.
+            Shape is (Nblts,). Ignored if `use_old_proj` is True.
+        cat_name : str
+            Name for the newly unprojected entry in the phase_center_catalog.
 
         Raises
         ------
         ValueError
-            If the phase_type is not 'phased'
+            If the object is alread unprojected.
         """
-        if self.phase_type == "phased":
-            pass
-        else:
-            raise ValueError(
-                "The data is already drift scanning; can only unphase phased data."
-            )
-
         if not use_old_proj:
-            telescope_location = self.telescope_location_lat_lon_alt
 
+            # TODO: this comment doesn't make sense to me. It's actually selection
+            #   *projected* blts not *unprojected* blts...
```

Comment:
There's a not operator `~` before the call to `_check_for_cat_type`, which means that the logic is opposite the logic in this comment. Thus my confusion. I added this TODO because I was confused. @kartographer can you weigh in here?

Reply:
Yeah, this wasn't the most clear comment -- the code below this (inside `unproject_phase`) basically checks for data that is already unprojected, because where that's true, nothing needs to be additionally done. Basically it's checking for blts require no modification (a kind of selective no-op check).

Reply:
ok, I'll reword the comment and take out the TODO.